apps/examples/testcase/le_tc/network/itc_net_connect.c

```diff
@@ -0,0 +1,269 @@
+/****************************************************************************
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/// @file itc_net_connect.c
+/// @brief ITC Scenario Test Case for connect() API
+
+#include <sys/socket.h>
+#include "tc_internal.h"
+
+#define SERVER_MSG "Hello from server"
+#define CLIENT_MSG "Hello from client 0 "
+#define BUFF_LEN 64
+#define SEC_2 2
+#define SEC_3 3
+#define SEC_7 7
+
+static int g_flag;
+static int g_port;
+
+/**
+* @fn               :server_connect
+* @description      :starts server
+* @return           :void
+*/
+void *server_connect(void *ptr_num_clients)
+{
+	int server_socket;
+	int client_socket;
+	int valread;
+	int opt = 1;
+	int ret;
+	struct sockaddr_in address;
+	int addrlen = sizeof(address);
+	int num_clients = *((int *)ptr_num_clients);
+	char buffer[BUFF_LEN] = {0};
+	char *ptr_msg = SERVER_MSG;
+
+	server_socket = socket(AF_INET, SOCK_STREAM, 0);
+	if (server_socket < 0) {
+		g_flag = ERROR;
+		return NULL;
+	}
+	setsockopt(server_socket, SOL_SOCKET, SO_REUSEADDR | SO_REUSEPORT, &opt, sizeof(opt));
+	address.sin_family = AF_INET;
+	address.sin_addr.s_addr = INADDR_ANY;
+	address.sin_port = htons(g_port);
+	ret = bind(server_socket, (struct sockaddr *)&address, sizeof(address));
+	if (ret != OK) {
+		g_flag = ERROR;
+		close(server_socket);
+		return NULL;
+	}
+	ret = listen(server_socket, 5);
+	if (ret != OK) {
+		g_flag = ERROR;
+		close(server_socket);
+		return NULL;
+	}
+	while (num_clients-- > 0) {
+		client_socket = accept(server_socket, (struct sockaddr *)&address, (socklen_t*)&addrlen);
+		if (client_socket < 0) {
+			g_flag = ERROR;
+			close(server_socket);
+			return NULL;
+		}
+		memset(buffer, 0, BUFF_LEN);
+		valread = read(client_socket, buffer, BUFF_LEN);
+		if (valread != strlen(CLIENT_MSG)) {
+			g_flag = ERROR;
+			close(client_socket);
+			close(server_socket);
+			return NULL;
+		}
+		strncpy(buffer + valread, ptr_msg, sizeof(SERVER_MSG));
```

Comment:
include header to use strncpy

Reply:
@sunghan-chang header files are included